tasks/shell_crmsh/create-and-push-cib.yml

```diff
@@ -0,0 +1,266 @@
+  ansible.builtin.tempfile:
+    state: file
+    suffix: _ha_cluster_original_cib_xml
+  register: __ha_cluster_tempfile_original_cib_xml
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+- name: Create a tempfile for new CIB
+  ansible.builtin.tempfile:
+    state: file
+    suffix: _ha_cluster_cib_xml
+  register: __ha_cluster_tempfile_cib_xml
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+# Maintenance mode is required, because CIB version changes with cluster
+# status changes, resulting in shadow CIB outdated and unable to patch.
+# Sleep is implemented to ensure that cluster have enough time to freeze
+# to ensure CIB export consistency.
+- name: Put cluster in maintenance mode to freeze cib changes
+  ansible.builtin.shell:
+    cmd: crm configure property maintenance-mode=true; sleep 10
+  check_mode: false
+  changed_when: true
+
+- name: Fetch CIB configuration
+  ansible.builtin.command:
+    cmd: cibadmin --query
+  register: __ha_cluster_fetch_cib
+  check_mode: false
+  changed_when: false  # this is a read-only command
+
+- name: Write CIB configuration
+  ansible.builtin.copy:
+    content: "{{ __ha_cluster_fetch_cib.stdout }}"
+    dest: "{{ item }}"
+    owner: root
+    group: root
+    mode: '0600'
+  loop:
+    - "{{ __ha_cluster_tempfile_cib_xml.path }}"
+    - "{{ __ha_cluster_tempfile_original_cib_xml.path }}"
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+# Starting with an empty CIB would remove all nodes and other parts of CIB
+# automatically created by pacemaker. That would effectively make the role to
+# report changed == True every time. Therefore, we start with the current
+# cluster CIB and purge it instead, thus keeping all the automatically created
+# parts in place and report only actual configuration changes.
+- name: Purge new CIB configuration
+  ansible.builtin.command:
+    cmd: >
+      cibadmin --force --delete-all --xpath
+      '/cib/configuration/*[not(
+        self::crm_config or
+        self::nodes or
+        self::resources or
+        self::constraints
+      )]
+      | /cib/configuration/*[self::resources or self::constraints]/*
+      | /cib/configuration/nodes/*/*
+      | /cib/configuration/crm_config//nvpair[not(
+        @name="cluster-infrastructure" or
+        @name="cluster-name" or
+        @name="dc-version" or
+        @name="have-watchdog" or
+        @name="last-lrm-refresh" or
+        @name="stonith-watchdog-timeout"
+      )]'
+  environment:
+    CIB_file: "{{ __ha_cluster_tempfile_cib_xml.path }}"
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+# Create/Replace shadow configuration with new based on current cluster.
+- name: Create new shadow crm configuration with force
+  ansible.builtin.command:
+    cmd: crm configure cib new {{ __ha_cluster_crm_shadow }} --force
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+# Build the new CIB
+- name: Build the new CIB
+  block:
```

Comment:
why use a `block` here?

Reply:
I assume to maintain structure equality between the `shell_pcs` and `shell_crmsh` subdirectories and code?

Ref: https://github.com/linux-system-roles/ha_cluster/blob/1.13.5/tasks/shell_pcs/create-and-push-cib.yml#L77-L87

Reply:
the pcs code uses a block because it defines `vars` that are scoped to the block:
```yaml
- name: Build the new CIB
  vars:
    __ha_cluster_pcs_cli_role:
      promoted: "{{
        ('pcmk.cib.roles.promoted-unpromoted' in __ha_cluster_pcs_capabilities)
        | ternary('Promoted', 'Master') }}"  # wokeignore:rule=master
      unpromoted: "{{
        ('pcmk.cib.roles.promoted-unpromoted' in __ha_cluster_pcs_capabilities)
        | ternary('Unpromoted', 'Slave') }}"  # wokeignore:rule=slave
  block:
```
you only need to use a `block` if 1) you are defining `vars` scoped to the block 2) you have several tasks you want to conditionally execute using a `when` and you don't want to put the `when` on all of those tasks 3) you want to catch errors using `rescue` and run cleanup using `always`

Reply:
As the code is functional, and to try to maintain the structure equality - can this be amendment be skipped? This is then clearer for future code review / maintenance, and can append with the block-scoped `vars` if needed in future? 

Minor alteration to remove the block and append in future if needed, so not something that needs debate. If you prefer clean, the block can be removed.